### ISSUE TITLE
Cannot debug with input() function in linux (Fedora 19). The externalConsole option seems doesn't work properly

### ISSUE THREAD

**Issue by elyuh** [DonJayamanne/pythonVSCode#152]
I cannot debug a python program with the input() function in it.
As far as I'm concerned, it is because visual studio code cannot receive input strings from the internal debug console. I tried to set the externalConsole option to true in the lanch.json but it didn't help. There was no external console opened when I started a debug session.
If somebody has the same problem and knows how to fix it, please give me your advice.
Thank you very much.

**Comment by DonJayamanne** [comment]
Hi @elyuh, i'll check this out soon. The externalConsole option only works on the Mac and Windows.
I'll look into adding this for Ubuntu soon, hopefully the solution is the same.

**Comment by elyuh** [comment]
Many thanks, DonJayamanne.
I tried to google and work it out but failed badly.

**Comment by DonJayamanne** [comment]
I believe I've managed to get this fixed. I'll get this released in the next build.

**Comment by elyuh** [comment]
That sounds awesome mate. Look forward to it. Thank you very much.

**Comment by DonJayamanne** [comment]
Unfortunately I'm getting a Permission denied exception in Ubuntu when launching the terminal using x-terminal-emulator.
Looks like this will take a little longer to resolve.
Hang in there.

**Comment by elyuh** [comment]
Thanks for the update, DonJayamanne.
I'm using the built-in terminal of VSCode to 'Run' those type of programs in the meantime.
Keep in touch.

**Comment by DonJayamanne** [comment]
Hopefully the VS Code team will make it possible to launch the program to be debugged in this integrated terminal window [#7034](https://github.com/Microsoft/vscode/issues/7034). This will make life a lot more easier.

**Comment by elyuh** [comment]
If it is the case then I think Code will be my pick every single day. Look forward to your news. Thank DonJayamanne.

**Comment by nathanblair** [comment]
Hi Don,

Wanted to throw in that I'm having (I believe) a similar/same/related issue. I'm noticing that no external console gets opened at all; I imagine this stems from your comment that the external terminal only launches on Mac and Windows and that you were having problems getting permissions to launch a terminal on linux (Ubuntu).

Here's exactly what I'm doing (very simple because I'm new to Python and so far your extension has been a blessing!)

I have a single python file in a "Project" folder just called 'testing.py'. The contents of 'testing.py' are:

```
print "Program started"

name = raw_input( "Enter something: " )
print "You entered", name
```

I've noticed that if I use the launch configuration for Python Console Window, the externalConsole property is set to 'true' and there is no 'RedirectOutput' debug option active. This configuration results in no external terminal being launched, and I can't find any output from a print statement in the VS Code's Debug Console nor VS Code's Output window.

If I use the straight Python configuration (the default settings that come with the extension), of which the property for 'externalConsole' is not present at all and the debug option 'RedirectOutput' is present, and execute the above testing.py file, the first print statement and the "Enter something: " prompt is printed to VS Code's Debug Console, but then the debugger exits. I'm unsure where input is/should be being gathered from. I don't know if its supposed to come in from the Debug Console input, or (I highly doubt this) an external terminal window that might be running, or VS Code's Integrated Terminal.

I'm available to help provide more details and troubleshoot so just let me know if I can be of more assistance. Should have mentioned this earlier but I'm running the new stock Linux Mint 18 64-bit, VS Code 1.3. I'll be following along with your progress and that of the VS Code team to see if I can help at all too.

**Comment by DonJayamanne** [comment]
@nathanblair , as you have rightly identified, the external terminal doesn't get displayed under linux, and that' why raw_input isn't working for you.
raw_input and similar input will work only when using the external terminal at this stage.

**Comment by DonJayamanne** [comment]
@elyuh, @nathanblair finally, fixed (7ce72ef72babbed157b2731c21e8d642005b9732)
Will be available in the next release, immediately after the release of the next version of VS Code (imminent).
